### ISSUE TITLE
processors.py need to be updated for Python 3.10+ for collections

### ISSUE THREAD

**Issue by stevefelt** [python-bonobo/bonobo#412]
I am using the latest version of bonobo with Python 3.10.2 or 3.10.1 (tried both) and the issue is with the processoprs.py file not being updated to handle the importing of collections.  The fix is the following code:
change:   
from collections import Iterable
to:
from collections.abc import Iterable

**Comment by stevefelt** [comment]
Hmm .. interesting.  I find that code change is in the file in master and dev .. but when I installed bonobo that file did not have the changes on my local.  Close this Issue if I am out of my mind :)

**Comment by hairypalm** [comment]
Hi @stevefelt (and @hartym).  Is this project abandoned?  This issue is important, but even with this issue there is another compatibility issue for Python 3.10 (collections.Sequence in https://github.com/python-bonobo/bonobo/pull/410).  Perhaps others.  I am interested in using this package as it's quite good, but if it's abandoned then how can it be revitalized?

**Comment by dat-linux** [comment]
looks like this project is dead

**Comment by hairypalm** [comment]
@dat-linux Yes.  Too bad.  Worth forking or just pick another?  Quite a large package to just abandon one day, esp since it was promoted quite a bit.  Any recommendations on something else?

**Comment by dat-linux** [comment]
Luigi looks good. It's more general purpose but it can be used for analytics. https://www.digitalocean.com/community/tutorials/how-to-build-a-data-processing-pipeline-using-luigi-in-python-on-ubuntu-20-04

**Comment by krummja** [comment]
I came across this repo when looking around for ETL tools, and it's definitely sad that such a robust project is just collecting dust. I was considering forking and picking up maintenance on it if there was interest, which there seems like there is. I'll be looking at upgrading the existing API to 3.10 over the coming few weeks.

**Comment by raphaelcicupo** [comment]
> I came across this repo when looking around for ETL tools, and it's definitely sad that such a robust project is just collecting dust. I was considering forking and picking up maintenance on it if there was interest, which there seems like there is. I'll be looking at upgrading the existing API to 3.10 over the coming few weeks.

@krummja any advances on this upgrade?
